autonomi/src/networking/driver/swarm_events.rs

```diff
@@ -167,4 +174,26 @@ impl NetworkDriver {
 
         Ok(())
     }
+
+    fn handle_identify_received_event(
+        &mut self,
+        peer_id: PeerId,
+        info: libp2p::identify::Info,
+    ) -> Result<(), NetworkDriverError> {
+        if info
+            .protocols
+            .contains(&StreamProtocol::new(KAD_STREAM_PROTOCOL_ID))
+        {
+            for listen_addr in info.listen_addrs {
+                self.swarm
+                    .behaviour_mut()
+                    .kademlia
+                    .add_address(&peer_id, listen_addr);
```

Comment:
`listen_addrs` is not a trusted source. There can be unreachable or random addresses in here. Also, if we were to do this we might want to add more restrictions to when we add these. E.g. only if they're also supporting our req/resp protocol.

Reply:
By the way, I'm unaware of the current mechanism of how we fill our RT, but we need to make sure this doesn't conflict with that existing mechanism.